experimental/dds/tree2/src/test/feature-libraries/contextuallyTyped.spec.ts

```diff
@@ -81,10 +86,94 @@ describe("ContextuallyTyped", () => {
 		const numberSequence = SchemaBuilder.fieldSequence(numberSchema);
 		const primaryObject = builder.object("numbers", { local: { [EmptyKey]: numberSequence } });
 		const schema = builder.intoDocumentSchema(numberSequence);
-		const mapTree = applyTypesFromContext(schema, new Set([primaryObject.name]), []);
+		const mapTree = applyTypesFromContext({ schema }, new Set([primaryObject.name]), []);
 		const expected: MapTree = { fields: new Map(), type: primaryObject.name, value: undefined };
 		assert.deepEqual(mapTree, expected);
 	});
 
+	describe("cursorFromContextualData adds globalFieldKey", () => {
+		it("for empty contextual data.", () => {
+			const builder = new SchemaBuilder("Identifier Domain");
+			const identifierSchema = builder.primitive("identifier", ValueSchema.String);
+			const identifierFieldSchema = builder.globalField(
+				"identifier",
+				SchemaBuilder.fieldValue(identifierSchema),
+			);
+			const nodeSchema = builder.object("node", {
+				global: [identifierFieldSchema] as const,
+			});
+
+			const nodeSchemaData = builder.intoDocumentSchema(
+				SchemaBuilder.fieldOptional(nodeSchema),
+			);
+			const contextualData: ContextuallyTypedNodeDataObject = {};
+
+			const identifierField = [
+				{
+					value: "x",
+					type: identifierSchema.name,
+					fields: new Map(),
+				},
+			];
+			const treeFromContextualData = jsonableTreeFromCursor(
+				cursorFromContextualData(
+					{
+						schema: nodeSchemaData,
+						fieldSource: () => (): MapTree[] => identifierField,
+					},
+					new Set([nodeSchema.name]),
+					contextualData,
+				),
+			);
+
+			assert.equal(treeFromContextualData.globalFields?.identifier[0].value, "x");
+		});
+
+		it("for nested contextual data with no global fields provided", () => {
```

Comment:
nit: the test above _also_ provides no global fields, so either update the name of the above test to say "...with no global fields provided" or remove "with no global fields provided" from the name of this test so that they are consistent.

Reply:
removed `fieldSource: () => undefined`, updated test title